src/web/factory.ts

```diff
@@ -46,26 +50,42 @@ export class TurboFactory extends TurboBaseFactory {
         }
       }
     } else if (privateKey) {
-      signer = new ArweaveSigner(privateKey);
+      if (token === 'solana') {
```

Comment:
this code is mostly duplicated, but the imports we're using DO matter for the build